etcd/src/main/scala/io/buoyant/etcd/Key.scala

```diff
@@ -0,0 +1,282 @@
+package io.buoyant.etcd
+
+import com.twitter.finagle.{Path, Service}
+import com.twitter.finagle.http._
+import com.twitter.io.Buf
+import com.twitter.util._
+
+case class BackoffsExhausted(key: Path, throwable: Throwable)
+  extends Exception(key.show, throwable)
+
+object Key {
+
+  /*
+   * Helpers for building request params
+   */
+
+  private val Params = Seq.empty[(String, String)]
+
+  private def trueParam(name: String, cond: Boolean): Option[(String, String)] =
+    if (cond) Some(name -> "true") else None
+
+  private def falseParam(name: String, cond: Boolean): Option[(String, String)] =
+    if (!cond) Some(name -> "false") else None
+
+  private[this] val dirParam = ("dir" -> "true")
+  private def dirOrValueParam(v: Option[Buf]): (String, String) = v match {
+    case None => dirParam
+    case Some(buf) =>
+      val Buf.Utf8(v) = buf
+      ("value" -> v)
+  }
+
+  private def ttlParam(ttl: Option[Duration]): Option[(String, String)] = ttl.map {
+    case Duration.Zero => "ttl" -> ""
+    case ttl => "ttl" -> ttl.inSeconds.toString
```

Comment:
tioli: I prefer to avoid shadowing, so maybe it would be good to use a name other than "ttl" here?